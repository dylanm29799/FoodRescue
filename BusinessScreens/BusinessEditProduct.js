import React, { useState, useEffect } from "react";
import {
  View,
  Text,
  StyleSheet,
  TextInput,
  FlatList,
  SafeAreaView,
  KeyboardAvoidingView,
} from "react-native";
import { scale } from "../components/ResponsiveText";
import Colour from "../constants/Colour";
import ButtonCustom from "../constants/ButtonCustom";
import * as firebase from "firebase";
import { ScrollView } from "react-native-gesture-handler";
import NormalProduct from "../components/NormalProduct";

const BusinessEditProduct = (props) => {
  const [itemName, setItemName] = useState("");
  const [quantity, setQuantity] = useState("");
  const [price, setPrice] = useState("");
  const [usualPrice, setUsualPrice] = useState("");
  const param = props.navigation.getParam("productKey");
  console.log(param);
  const dbconnection = firebase.firestore();
  var user = firebase.auth().currentUser;
  var docRef = dbconnection.collection("Products").doc(param);
  const handleSubmit = () => {
    docRef
      .get()
      .then(function (doc) {
        if (doc.exists) {
          alert("Worked");
          console.log(itemName, quantity, price, usualPrice);
          return docRef.update({
            itemName: itemName,
            quantity: quantity,
            newPrice: price,
            usualPrice: usualPrice,
          });
        }
      })
      .then(function () {
        console.log("Document successfully updated!");
      })
      .catch(function (error) {
        // The document probably doesn't exist.
        console.error("Error updating document: ", error);
      });
  };
  useEffect(() => {
    if (user) {
      // User is signed in.

      console.log(user.uid);
      docRef
        .get()
        .then(function (doc) {
          if (doc.exists) {
            setItemName(doc.data().itemName);
            setQuantity(doc.data().quantity);
            setPrice(doc.data().newPrice);
            setUsualPrice(doc.data().usualPrice);
          } else {
            // doc.data() will be undefined in this case
            console.log("No such document!");
          }
        })

        .catch(function (error) {
          console.log("Error getting document:", error);
        });
    } else {
      // No user is signed in.
      console.log("No User Logged In");
    }
  }, []);

  return (
    <View
      behavior={Platform.OS === "ios" ? "padding" : "height"}
      style={styles.screen}
    >
      <View
        style={{
          width: "100%",
          justifyContent: "center",
          paddingBottom: scale(30),
        }}
      >
        <Text style={styles.all}> Product Name:</Text>
        <TextInput
          style={styles.textInput}
          onChangeText={(itemName) => setItemName(itemName)}
        >
          {itemName}
        </TextInput>
        <Text style={styles.all}> Quantity:</Text>
        <TextInput
          style={styles.textInput}
          onChangeText={(quantity) => setQuantity(quantity)}
        >
          {quantity}
        </TextInput>

        <Text style={styles.all}> Original Price:</Text>
        <TextInput
          style={styles.textInput}
          onChangeText={(usualPrice) => setUsualPrice(usualPrice)}
        >
          {usualPrice}
        </TextInput>

        <Text style={styles.all}> Price of the Product:</Text>
        <TextInput
          style={styles.textInput}
          onChangeText={(price) => setPrice(price)}
        >
          {price}
        </TextInput>
      </View>
      <ButtonCustom onPress={handleSubmit} title="Submit" />
    </View>
  );
};

const styles = StyleSheet.create({
  screen: {
    flex: 1,
    justifyContent: "center",
    alignItems: "center",
    width: "70%",
    alignSelf: "center",
  },
  textInput: {
    borderColor: Colour.primaryColour,
    paddingLeft: 15,
    borderRadius: 5,
    borderWidth: 2,
    justifyContent: "center",
    padding: scale(3),
    fontFamily: "MonM",
    fontSize: scale(15),
  },

  all: {
    paddingTop: scale(10),
    fontFamily: "MonM",
    fontSize: scale(12),
  },
});

export default BusinessEditProduct;
