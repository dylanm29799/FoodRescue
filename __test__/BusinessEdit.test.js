// __tests__/BusinessEdit.test.js

import React from "react";
import BusinessEditProduct from "../BusinessScreens/BusinessEditProduct";
import renderer from "react-test-renderer";
import { firebaseConfig } from "../config";
import * as firebase from "firebase";

firebase.initializeApp(firebaseConfig);

test("renders correctly", () => {
  const tree = renderer.create(<BusinessEditProduct />).toJSON();
  expect(tree).toMatchSnapshot();
});
