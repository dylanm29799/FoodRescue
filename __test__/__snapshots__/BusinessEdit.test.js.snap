// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "alignSelf": "center",
      "flex": 1,
      "justifyContent": "center",
      "width": "70%",
    }
  }
>
  <View
    style={
      Object {
        "justifyContent": "center",
        "paddingBottom": 42.857142857142854,
        "width": "100%",
      }
    }
  >
    <Text
      style={
        Object {
          "fontFamily": "MonM",
          "fontSize": 25.714285714285715,
          "paddingTop": 10.714285714285714,
        }
      }
    >
       Product Name:
    </Text>
    <TextInput
      allowFontScaling={true}
      onChangeText={[Function]}
      rejectResponderTermination={true}
      style={
        Object {
          "borderColor": "#c44536",
          "borderRadius": 5,
          "borderWidth": 2,
          "fontFamily": "MonM",
          "fontSize": 32.14285714285714,
          "justifyContent": "center",
          "padding": 6.428571428571429,
          "paddingLeft": 15,
        }
      }
      underlineColorAndroid="transparent"
    >
      
    </TextInput>
    <Text
      style={
        Object {
          "fontFamily": "MonM",
          "fontSize": 25.714285714285715,
          "paddingTop": 10.714285714285714,
        }
      }
    >
       Quantity:
    </Text>
    <TextInput
      allowFontScaling={true}
      onChangeText={[Function]}
      rejectResponderTermination={true}
      style={
        Object {
          "borderColor": "#c44536",
          "borderRadius": 5,
          "borderWidth": 2,
          "fontFamily": "MonM",
          "fontSize": 32.14285714285714,
          "justifyContent": "center",
          "padding": 6.428571428571429,
          "paddingLeft": 15,
        }
      }
      underlineColorAndroid="transparent"
    >
      
    </TextInput>
    <Text
      style={
        Object {
          "fontFamily": "MonM",
          "fontSize": 25.714285714285715,
          "paddingTop": 10.714285714285714,
        }
      }
    >
       Original Price:
    </Text>
    <TextInput
      allowFontScaling={true}
      onChangeText={[Function]}
      rejectResponderTermination={true}
      style={
        Object {
          "borderColor": "#c44536",
          "borderRadius": 5,
          "borderWidth": 2,
          "fontFamily": "MonM",
          "fontSize": 32.14285714285714,
          "justifyContent": "center",
          "padding": 6.428571428571429,
          "paddingLeft": 15,
        }
      }
      underlineColorAndroid="transparent"
    >
      
    </TextInput>
    <Text
      style={
        Object {
          "fontFamily": "MonM",
          "fontSize": 25.714285714285715,
          "paddingTop": 10.714285714285714,
        }
      }
    >
       Price of the Product:
    </Text>
    <TextInput
      allowFontScaling={true}
      onChangeText={[Function]}
      rejectResponderTermination={true}
      style={
        Object {
          "borderColor": "#c44536",
          "borderRadius": 5,
          "borderWidth": 2,
          "fontFamily": "MonM",
          "fontSize": 32.14285714285714,
          "justifyContent": "center",
          "padding": 6.428571428571429,
          "paddingLeft": 15,
        }
      }
      underlineColorAndroid="transparent"
    >
      
    </TextInput>
  </View>
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#c44536",
        "borderRadius": 10,
        "height": 50,
        "justifyContent": "center",
        "opacity": 1,
        "padding": 10,
      }
    }
  >
    <Text
      style={
        Object {
          "alignItems": "center",
          "color": "#EBF38B",
          "fontFamily": "MonM",
          "fontSize": 38.57142857142857,
          "textAlign": "center",
          "width": 321.42857142857144,
        }
      }
    >
      Update
    </Text>
  </View>
</View>
`;
